pupil_src/shared_modules/video_capture/fake_backend.py

```diff
@@ -34,8 +34,6 @@ def __init__(self, timestamp, img, index):
         self._gray = None
         self.index = index
         # indicate that the frame does not have a native yuv or jpeg buffer
-        self.yuv_buffer = None
```

Comment:
why not make these required form the frame api? Then you dont have to check for attributes...

Reply:
Frame publisher's incompatible format detection is based on AttributeErrors. We need to check if this attribute is `None` on multiple occasions anyway. My thought was that we could change these to an existence test of the attribute and keep the simple fail logic in frame publisher. Alternatively, we always define `yuv_buffer`, need to check multiple times for `None` and increase frame publishers fail logic complexity.